### ISSUE TITLE
How to know the number of classes in a custom trained model

### ISSUE THREAD

**Issue by grhaonan** [ultralytics/yolov5#9485]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi guys,

Is there any simple way to fetch the number of classes/labels from a custom trained YOLO model?

For example, if model A is a trained model based on YOLO with a custom dataset which has two classes/labels (c1,c2)

Any way to extract the number of classes/labels of this model (2 in this example) after loading this custom trained model through torch.hub.load()? Hugging face offers a method called num_labels to return this value but I didn't find a similar function from torch model object or any YOLO document.

Thanks

**Comment by hardikdava** [comment]
@grhaonan you can check the output of the network. The shape of the model output is (1, numbers of detections, numbers of class + 5). So for a model trained for 2 classes the shape of the output array will be (1, numbers of detections, 7), and for the coco pretrained model with 80 classes gives an output of shape (1, numbers of detections, 85). The numbers of detections depends on numbers of anchors.

**Comment by glenn-jocher** [comment]
@grhaonan yes of course:

<img width="906" alt="Screenshot 2022-09-19 at 12 03 43" src="https://user-images.githubusercontent.com/26833433/190994504-4b500953-46e4-4b68-9097-56ef165e87e0.png">

**Comment by grhaonan** [comment]
> @grhaonan yes of course:
> 
> <img alt="Screenshot 2022-09-19 at 12 03 43" width="906" src="https://user-images.githubusercontent.com/26833433/190994504-4b500953-46e4-4b68-9097-56ef165e87e0.png">

Big thanks for your quick reply, this is very handy and what I looked for :)

**Comment by glenn-jocher** [comment]
@grhaonan you're welcome! I'm glad that was helpful. If you have any more questions about YOLOv5 or anything else, feel free to ask. Thank you for your interest!
